src/clj_jargon/permissions.clj

```diff
@@ -570,15 +585,43 @@
        cm (.getPath src) (.getPath dst) user admin-users skip-source-perms?))))
 
 (defn permissions
-  [cm user fpath]
+  [cm user fpath & {:keys [known-type] :or {known-type nil}}]
   (validate-path-lengths fpath)
-  (case (item/object-type cm fpath)
+  (case (or known-type (item/object-type cm fpath))
     :dir  (collection-perm-map cm user fpath)
     :file (dataobject-perm-map cm user fpath)
     {:read false
      :write false
      :own false}))
 
+(defn- larger-perm
+  [perm1 perm2]
+  (cond
+    (nil? perm1)
+    perm2
+
+    (nil? perm2)
+    perm1
+
+    (> (perm-order-map perm1) (perm-order-map perm2))
+    perm1
```

Comment:
Seems like the combination of the next condition and the `:else` case already covers this, so this condition could be removed 🤔

Reply:
Hm, probably. I should look into the `take-upto` thing I mentioned and then forgot about, because I think with that I can use `(partial = own-perm)` instead of `check-more-perms?` and this, and then it's somewhat moot, lol